src/scripts/ui/PlayerTradeComponent.tsx

```diff
@@ -29,6 +125,9 @@ export function PlayerTradeComponent({ gameState, xy }: IBuildingComponentProps)
    const user = useUser();
    const myXy = getMyMapXy();
 
+   const [wantFilter, setWantFilter] = useState<keyof ResourceDefinitions|string>("");
```

Comment:
Again, consider using a `Set`. If only single filter can be supported. the signature should be like `useState<Resource|null>`. Empty string is a bad item here to express null.

Reply:
Simplified useStates and tidied up types.